feat: add `blas/base/wasm/sdsdot`

ShabiShett07 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Progresses #2039

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds the package `blas/base/wasm/sdsdot`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add blas/base/wasm/sdsdot

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/README.md:
```diff
@@ -0,0 +1,326 @@
+The function has the following parameters:
+
+-   **N**: number of indexed elements.
+-   **scalar**: scalar constant to add to dot product
```
```suggestion
-   **scalar**: scalar constant to add to dot product.
```

kgryte committed:
docs: add missing period

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/README.md:
```diff
@@ -0,0 +1,326 @@
+The function has the following parameters:
+
+-   **N**: number of indexed elements.
+-   **scalar**: scalar constant to add to dot product
```
```suggestion
-   **scalar**: scalar constant to add to dot product.
```

kgryte committed:
docs: fix missing period

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/repl.txt:
```diff
@@ -0,0 +1,557 @@
+    Indexing is relative to the first index. To introduce an offset, use typed
+    array views.
+
+    If `N <= 0`, both the functions return `0`.
```
```suggestion
    If `N <= 0`, the function returns `0`.
```

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/repl.txt:
```diff
@@ -0,0 +1,557 @@
+
+{{alias}}.Module.prototype.ndarray( N, scalar, xp, sx, ox, yp, sy, oy )
+    Computes the dot product of two single-precision floating-point vectors with
+    extended accumulation, using alternative indexing semantics.
```
```suggestion
    extended accumulation using alternative indexing semantics.
```

kgryte committed:
docs: use description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/repl.txt:
```diff
@@ -0,0 +1,557 @@
+    > var out = mod.ndarray( 5, 0.0, xptr, 1, 0, yptr, 1, 0 )
+    15.0
+
+
```
```suggestion
```

kgryte committed:
docs: remove blank line

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/types/test.ts:
```diff
@@ -0,0 +1,577 @@
+	sdsdot.ndarray( x.length, 0.0, x, 1, 0, y, ( x: number ): number => x, 0 ); // $ExpectError
+}
+
+// The compiler throws an error if the `ndarray` method is provided a eighth argument which is not a number...
```
```suggestion
// The compiler throws an error if the `ndarray` method is provided an eighth argument which is not a number...
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/types/test.ts:
```diff
@@ -0,0 +1,577 @@
+	});
+	const mod = sdsdot.Module( mem );
+
+	mod.ndarray( 10, '10', 1, 0, 80, 1, 0 ); // $ExpectError
```
You're missing an argument. You are only providing seven arguments and there are eight.

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/types/test.ts:
```diff
@@ -0,0 +1,577 @@
+	});
+	const mod = sdsdot.Module( mem );
+
+	mod.ndarray( 10, 0, '10', 0, 80, 1, 0 ); // $ExpectError
```
You're missing an argument. You are only providing seven arguments and there are eight.

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/types/test.ts:
```diff
@@ -0,0 +1,577 @@
+	mod.ndarray( 10, 0, ( x: number ): number => x, x, 0, 80, 1, 0 ); // $ExpectError
+}
+
+// The compiler throws an error if the `ndarray` method of a module instance is provided a fourth argument which is not a number...
```
```suggestion
// The compiler throws an error if the `ndarray` method of a module instance is provided a fifth argument which is not a number...
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/types/test.ts:
```diff
@@ -0,0 +1,577 @@
+	mod.ndarray( 10, 0, 0, 1, ( x: number ): number => x, 80, 1, 0 ); // $ExpectError
+}
+
+// The compiler throws an error if the `ndarray` method of a module instance is provided a fifth argument which is not a number...
```
```suggestion
// The compiler throws an error if the `ndarray` method of a module instance is provided a sixth argument which is not a number...
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/types/test.ts:
```diff
@@ -0,0 +1,577 @@
+	mod.ndarray( 10, 0, 0, 1, 0, ( x: number ): number => x, 0, 1, 0 ); // $ExpectError
+}
+
+// The compiler throws an error if the `ndarray` method of a module instance is provided a sixth argument which is not a number...
```
```suggestion
// The compiler throws an error if the `ndarray` method of a module instance is provided a seventh argument which is not a number...
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/types/test.ts:
```diff
@@ -0,0 +1,577 @@
+	mod.ndarray( 10, 0, 0, 1, 0, 80, ( x: number ): number => x, 0 ); // $ExpectError
+}
+
+// The compiler throws an error if the `ndarray` method of a module instance is provided a seventh argument which is not a number...
```
```suggestion
// The compiler throws an error if the `ndarray` method of a module instance is provided an eighth argument which is not a number...
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/package.json:
```diff
@@ -0,0 +1,80 @@
+{
+  "name": "@stdlib/blas/base/wasm/sdsdot",
+  "version": "0.0.0",
+  "description": "Compute the dot product of `x` and `y` with extended accumulation and result.",
```
```suggestion
  "description": "Calculate the dot product of two single-precision floating-point vectors with extended accumulation.",
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/test/test.main.js:
```diff
@@ -0,0 +1,229 @@
+	t.end();
+});
+
+tape( 'the `main` method computes the dot product of `x` and `y` with extended accumulation and result', function test( t ) {
```
```suggestion
tape( 'the `main` method computes the dot product of `x` and `y` with extended accumulation', function test( t ) {
```

kgryte replied:
`sdsdot` -> sds -> single-precision in, double-precision accumulate, single-precision out. The result is not in extended precision.

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/test/test.module.main.js:
```diff
@@ -0,0 +1,283 @@
+	t.end();
+});
+
+tape( 'a module instance has a `main` method which computes the dot product of `x` and `y` with extended accumulation and result', function test( t ) {
```
```suggestion
tape( 'a module instance has a `main` method which computes the dot product of `x` and `y` with extended accumulation', function test( t ) {
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/test/test.module.ndarray.js:
```diff
@@ -0,0 +1,360 @@
+	t.end();
+});
+
+tape( 'a module instance has an `ndarray` method which computes the dot product of `x` and `y` with extended accumulation and result', function test( t ) {
```
```suggestion
tape( 'a module instance has an `ndarray` method which computes the dot product of `x` and `y` with extended accumulation', function test( t ) {
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/sdsdot/test/test.ndarray.js:
```diff
@@ -0,0 +1,251 @@
+	t.end();
+});
+
+tape( 'the `ndarray` method computes the dot product of `x` and `y` with extended accumulation and result', function test( t ) {
```
```suggestion
tape( 'the `ndarray` method computes the dot product of `x` and `y` with extended accumulation', function test( t ) {
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: fix description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (changes requested):
Other than the TypeScript test file, this is looking good after some clean-up.

ShabiShett07 committed:
chore: add correct argument number

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add correct argument number

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, I think now it is ready for review

kgryte reviewed (approved):
LGTM
